kubeflow-pipelines/docling-vlm/docling_convert_components.py

```diff
@@ -0,0 +1,288 @@
+            print(f"import-test-pdfs: downloading {url} -> {dest}", flush=True)
+            with requests.get(url, stream=True, timeout=30) as resp:
+                resp.raise_for_status()
+                with dest.open("wb") as f:
+                    for chunk in resp.iter_content(chunk_size=8192):
+                        if chunk:
+                            f.write(chunk)
+    print("import-test-pdfs: done", flush=True)
+
+@dsl.component(
+    base_image=PYTHON_BASE_IMAGE,
+)
+def create_pdf_splits(
+    input_path: dsl.Input[dsl.Artifact],
+    num_splits: int,
+) -> List[List[str]]:
+    """
+    Create a list of PDF splits.
+
+    Args:
+        input_path: Path to the input directory containing PDF files.
+        num_splits: Number of splits to create.
+    """
+    from pathlib import Path # pylint: disable=import-outside-toplevel  # noqa: PLC0415, E402
+
+    input_path_p = Path(input_path.path)
+
+    all_pdfs = [path.name for path in input_path_p.glob("*.pdf")]
+    all_splits = [all_pdfs[i::num_splits] for i in range(num_splits)]
+    filled_splits = list(filter(None, all_splits))
+    return filled_splits
+
+
+@dsl.component(
+    base_image=DOCLING_BASE_IMAGE,
+)
+def download_docling_models(
+    output_path: dsl.Output[dsl.Artifact],
+    remote_model_endpoint_enabled: bool,
+):
+    """
+    Download Docling models.
+
+    Args:
+        output_path: Path to the output directory for Docling models.
+    """
+    from pathlib import Path # pylint: disable=import-outside-toplevel  # noqa: PLC0415, E402
+    from docling.utils.model_downloader import download_models # pylint: disable=import-outside-toplevel  # noqa: PLC0415, E402
+
+    output_path_p = Path(output_path.path)
+
+    output_path_p.mkdir(parents=True, exist_ok=True)
+
+    if remote_model_endpoint_enabled:
+        # Only models set are what lives in fabianofranz repo
+        # TODO: figure out what needs to be downloaded or removed
+        download_models(
+            output_dir=output_path_p,
+            progress=False,
+            force=False,
+            with_layout=True,
+            with_tableformer=True,
+            with_code_formula=False,
+            with_picture_classifier=False,
+            with_smolvlm=False,
+            with_smoldocling=False,
+            with_smoldocling_mlx=False,
+            with_granite_vision=False,
+            with_easyocr=False,
+        )
+    else:
+        download_models(
+            output_dir=output_path_p,
+            with_smolvlm=True,
+            with_smoldocling=True,
+            progress=False,
+            force=False,
+            with_layout=False,
+            with_tableformer=False,
+            with_code_formula=False,
+            with_picture_classifier=False,
+            with_smoldocling_mlx=False,
+            with_granite_vision=False,
+            with_easyocr=False,
+        )
+
+
+@dsl.component(
+    base_image=DOCLING_BASE_IMAGE,
+)
+def docling_convert(
```

Comment:
Do `pipeline` or `vlm-model` need to be passed here or is it enough to pass the right vlm model?